src/java.desktop/share/classes/java/awt/image/Raster.java

```diff
@@ -297,11 +330,18 @@ public static WritableRaster createInterleavedRaster(int dataType,
      * @param location  the upper-left corner of the {@code Raster}
      * @return a WritableRaster object with the specified data type,
      *         width, height and number of bands.
-     * @throws RasterFormatException if {@code w} or {@code h}
-     *         is less than or equal to zero, or computing either
+     * @throws IllegalArgumentException if {@code dataType} is not
+     *         one of the supported data types, which are
+     *         {@code DataBuffer.TYPE_BYTE},
+     *         {@code DataBuffer.TYPE_USHORT}
+     *         or {@code DataBuffer.TYPE_INT}
```

Comment:
What about other types like TYPE_SHORT or we assume it is unsupported?

Reply:
The existing code throws IAE if it is not one of these exact types. See around line 460 in this updated file.
And that code is in a public factory method called by this factory method. And that callee has exactly this documentation. So I am just copying it over so that it is consistently documented.